exercises/run-length-encoding/src/example/java/RunLengthEncoding.java

```diff
@@ -0,0 +1,47 @@
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class RunLengthEncoding {
+    public String encode(String data) {
+        String encodedData = "";
+
+        for (int i = 0; i < data.length(); i++) {
+            char c = data.charAt(i);
+
+            if (i != 0 && c == data.charAt(i - 1)) continue;
+
+            int amt = 1;
+
+            for (int j = i + 1; j < data.length() && c == data.charAt(j); j++) {
+                amt++;
+            }
+
+            if (amt == 1) encodedData += c;
+            else encodedData += amt + Character.toString(c);
```

Comment:
Cool!  I've never heard about StringBuilder before, it seems like it would be a lot better though considering that concatenating a string creates a `new` string every time